jump-leveleditor/src/main/java/com/bitdecay/jump/leveleditor/render/mouse/DropObjectMode.java

```diff
@@ -31,7 +31,8 @@ public void setObject(Class<? extends RenderableLevelObject> objectClass) {
 
     private void getNewReference() {
         try {
-            reference = objectClass.newInstance();
+            if (reference != null) reference = reference.newInstance();
+            else reference = objectClass.newInstance();
```

Comment:
I can't say I follow this entirely. It feels a little weird that we have two ways of getting a new instance of a given class. Why do we have both a class level and an instance level function?